src/schematools/types.py

```diff
@@ -37,7 +50,7 @@ def json_data(self) -> Dict[str, Any]:
         return self.data
 
     @classmethod
-    def from_dict(cls, obj: Dict[str, Any]) -> ST:
+    def from_dict(cls, obj: Dict[str, Any]) -> SchemaType:
```

Comment:
This is incorrect. `SchemaType.from_dict` is inherited by more specialized classes, eg `DatasetSchema`. If we call `DatasetSchema.from_dict` we do **not** want it to return a `SchemaType`, we want it to return a `DatasetSchema`. The `TypeVar` `ST` ensure exactly that.

Reply:
I'm sorry. Ignore my comment. `-> SchemaType` is correct.